### PR TITLE
[WEB-2098] logged out page and 401/403 redirects

### PR DESCRIPTION
**Pull request by krystophv**
for [WEB-2098] creates logged out page which users will be sent to when keycloak token authentication/refresh fails. Additionally triggers a keycloak token refresh whenever any API call returns a 401/403 error to verify session validity.

[WEB-2098]: https://tidepool.atlassian.net/browse/WEB-2098?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/unit/keycloak.test.js [hunk 1]
```diff
@@ -264,6 +266,37 @@ describe('keycloak', () => {
       expect(updateKeycloakConfigMock.callCount).to.equal(0);
       keycloak.__ResetDependency__('_keycloakConfig');
     });
+
+    it('should call keycloak.updateToken() when action has 401 or 403 error', () => {
```
Comment (tjotala) [inline test/unit/keycloak.test.js:270]: Stylistic nit: these might be "cleaner" as two separate test cases.

File: app/keycloak.js [hunk 2]
```diff
@@ -103,6 +105,16 @@ export const keycloakMiddleware = (api) => (storeAPI) => (next) => (action) => {
       break;
     }
     default:
+      if (
+        action?.error?.status === 401 ||
+        action?.error?.originalError?.status === 401 ||
+        action?.error?.status === 403 ||
```
Comment (tjotala) [inline app/keycloak.js:111]: I can see how this makes perfect sense for 401.

More of a philosophical question here: will re-login solve a 403?

I suppose won't hurt to try, but if they didn't have the permission before, it seems unlikely that they would gain it after logging in again - unless someone just granted them the permission.

**Review by tjotala — APPROVED** [review]
LGTM - no blockers, just nits

File: app/keycloak.js (hunk 2 above)
Comment (krystophv) [reply app/keycloak.js:111]: There's some amount of historical ambiguity around which status the backend returns in which situation. There have been cases where we should have used one but return the other, so this code treats both the same way.

File: test/unit/keycloak.test.js (hunk 1 above)
Comment (krystophv) [reply test/unit/keycloak.test.js:270]: Wouldn't hurt to split this out. 👍🏼

**Pull request merged**